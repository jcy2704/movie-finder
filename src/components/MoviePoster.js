import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import GenreList from '../helpers/genres/genre';

const MoviePoster = ({ movie }) => {
  const {
    id, poster_path: poster, title, vote_average: rating, genre_ids: genreIds,
  } = movie;

  const history = useHistory();

  const switcher = () => {
    history.push(`/movie/${id}`);
  };

  return (
    <li className="single-poster">
      <div role="presentation" onClick={switcher} className="movie position-relative">
        <div className="poster-img">
          <img src={`${process.env.REACT_APP_POSTER_URL}${poster}`} alt="Poster" />
        </div>
        <div className="w-100 info-cont-catalogue position-absolute">
          <div className="info-top d-flex justify-content-between">
            <h3>
              {title}
            </h3>
            <p className="catalogue-rating">
              <FontAwesomeIcon className="rating-star" icon={faStar} />
              {rating}
            </p>
          </div>
          <ul className="catalogue-genres flex-wrap d-flex">
            {GenreList(genreIds).map(genre => (
              <li key={genre}>
                <button type="button" name={genre}>
                  {genre}
                </button>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </li>
  );
};

MoviePoster.propTypes = {
  movie: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default MoviePoster;
