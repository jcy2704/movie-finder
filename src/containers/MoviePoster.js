import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import GenreList from '../helpers/genres/genre';
import MovieModal from '../components/MovieModal';
import getVideo from '../helpers/getVideo';

const MoviePoster = ({ movie }) => {
  const [showModal, setShowModal] = useState(false);
  const [videoURL, setVideoURL] = useState({
    key: '',
  });

  const {
    id, poster_path: poster, title, vote_average: rating, genre_ids: genreIds,
  } = movie;

  const switcher = () => {
    setShowModal(!showModal);
    getVideo(id, setVideoURL);
  };

  return (
    <li className="single-poster">
      <div role="presentation" onClick={switcher} className="movie position-relative">
        <div className="poster-img">
          <img src={`${process.env.REACT_APP_POSTER_URL}${poster}`} alt="Poster" />
        </div>
        <div className="w-100 info-cont-catalogue position-absolute">
          <div className="info-top d-flex justify-content-between">
            <h3>
              {title}
            </h3>
            <p className="catalogue-rating">
              <FontAwesomeIcon className="rating-star" icon={faStar} />
              {rating}
            </p>
          </div>
          <ul className="catalogue-genres flex-wrap d-flex">
            {GenreList(genreIds).map(genre => (
              <li key={genre}>
                <button type="button" name={genre}>
                  {genre}
                </button>
              </li>
            ))}
          </ul>
        </div>
      </div>
      <MovieModal showModal={showModal} switcher={switcher} movie={movie} video={videoURL} />
    </li>
  );
};

MoviePoster.propTypes = {
  movie: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default MoviePoster;
