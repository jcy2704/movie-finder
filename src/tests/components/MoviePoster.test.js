import React from 'react';
import renderer from 'react-test-renderer';
import MoviePoster from '../../components/MoviePoster';

const movie = {
  backdrop_path: '/inJjDhCjfhh3RtrJWBmmDqeuSYC.jpg',
  genre_ids: [
    28,
    878,
  ],
  id: 399566,
  overview: 'In a time when monsters walk the Earth, humanityâ€™s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.',
  poster_path: '/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
  title: 'Godzilla vs. Kong',
  vote_average: 8.4,
  video: {},
};

it('Renders Movies Posters', () => {
  const popular = renderer
    .create(
      <MoviePoster movie={movie} />,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});
