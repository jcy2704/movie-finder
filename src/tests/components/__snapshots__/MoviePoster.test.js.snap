// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Movies Posters 1`] = `
<li
  className="single-poster"
>
  <div
    className="movie position-relative"
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="poster-img"
    >
      <img
        alt="Poster"
        src="https://image.tmdb.org/t/p/w342/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
      />
    </div>
    <div
      className="w-100 info-cont-catalogue position-absolute"
    >
      <div
        className="info-top d-flex justify-content-between"
      >
        <h3>
          Godzilla vs. Kong
        </h3>
        <p
          className="catalogue-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 rating-star"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          8.4
        </p>
      </div>
      <ul
        className="catalogue-genres flex-wrap d-flex"
      >
        <li>
          <button
            name="Action"
            type="button"
          >
            Action
          </button>
        </li>
        <li>
          <button
            name="Science Fiction"
            type="button"
          >
            Science Fiction
          </button>
        </li>
      </ul>
    </div>
  </div>
</li>
`;
